src/type-exercises.ts

```diff
@@ -0,0 +1,249 @@
+} = client
+const sendingClient: toSendClient = {
+  name: clientName,
+  lastname,
+  emailAddres,
+  birthDate,
+  active,
+  verified,
+}
+
+// Now we can do something (like calling an API, or something like that) only with the general dataa
+const doSomething = (client: toSendClient) => {
+  console.log(client)
+}
+doSomething(sendingClient)
+/**
+ * Exercise #2: Implement the utility type `If<C, T, F>`, which evaluates a condition `C`
+ * and returns one of two possible types:
+ * - `T` if `C` is `true`
+ * - `F` if `C` is `false`
+ *
+ * @description
+ * - `C` is expected to be either `true` or `false`.
+ * - `T` and `F` can be any type.
+ *
+ * @example
+ * type A = If<true, 'a', 'b'>;  // expected to be 'a'
+ * type B = If<false, 'a', 'b'>; // expected to be 'b'
+ */
+
+// Add here your solution
+type If<C extends boolean, T, F> = C extends true ? T : F
+
+// Usage examples:
+type ClientWithoutCards = Omit<
+  Client,
+  'activeCreditCard' | 'alternativeCreditCard'
+>
+type ConditionalClient<C extends boolean> = If<C, Client, ClientWithoutCards>
+type VerifiedClient = ConditionalClient<true>
+type UnverifiedClient = ConditionalClient<false>
+
+// Add here your example
+
+/**
+ * Exercise #3: Recreate the built-in `Readonly<T>` utility type without using it.
+ *
+ * @description
+ * Constructs a type that makes all properties of `T` readonly.
+ * This means the properties of the resulting type cannot be reassigned.
+ *
+ * @example
+ * interface Todo {
+ *   title: string;
+ *   description: string;
+ * }
+ *
+ * const todo: MyReadonly<Todo> = {
+ *   title: "Hey",
+ *   description: "foobar"
+ * };
+ *
+ * todo.title = "Hello";       // Error: cannot reassign a readonly property
+ * todo.description = "barFoo"; // Error: cannot reassign a readonly property
+ */
+
+// Add here your solution
+type MyReadonly<T> = {
+  readonly [P in keyof T]: T[P]
+}
+
+type ReadOnlyClient = MyReadonly<Client>
+const readClient = client as ReadOnlyClient
+
+console.log(readClient)
+
+//Now we can't modify any property of readClient
+//readClient.active = false
+
+// Add here your example
+
+/**
+ * Exercise #4: Recreate the built-in `ReturnType<T>` utility type without using it.
+ *
+ * @description
+ * The `MyReturnType<T>` utility type extracts the return type of a function type `T`.
+ *
+ * @example
+ * const fn = (v: boolean) => {
+ *   if (v) {
+ *     return 1;
+ *   } else {
+ *     return 2;
+ *   }
+ * };
+ *
+ * type a = MyReturnType<typeof fn>; // expected to be "1 | 2"
+ */
+
+// Add here your solution
+type MyReturnType<F> = F extends (...args: any) => infer R ? R : any
+const readOrEditClient = (operation: 'read' | 'edit') => {
+  if (operation === 'edit') {
+    const verifiedClient: VerifiedClient = client as VerifiedClient
+    return verifiedClient
+  } else {
+    const unverifiedClient: UnverifiedClient = client as UnverifiedClient
+    return unverifiedClient
+  }
+}
+
+type functionType = MyReturnType<typeof readOrEditClient>
```

Comment:
Good solution

Reply:
It was solved in https://github.com/ravnhq/TS-Nederdy-Challanges/pull/52/commits/6e00443fb036a73c0e21719b97568611b533e1ae